pkg/logs/input/tailer/file_provider.go

```diff
@@ -0,0 +1,66 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2017 Datadog, Inc.
+
+package tailer
+
+import (
+	"log"
+	"path/filepath"
+
+	"github.com/DataDog/datadog-agent/pkg/logs/config"
+)
+
+// File represents a file to tail
+type File struct {
+	Path   string
+	Source *config.IntegrationConfigLogSource
+}
+
+// NewFile returns a new File
+func NewFile(path string, source *config.IntegrationConfigLogSource) *File {
+	return &File{
+		Path:   path,
+		Source: source,
+	}
+}
+
+// FileProvider implements the logic to retrieve at most filesLimit Files defined in sources
+type FileProvider struct {
+	sources    []*config.IntegrationConfigLogSource
+	filesLimit int
+}
+
+// NewFileProvider returns a new FileProvider
+func NewFileProvider(sources []*config.IntegrationConfigLogSource, filesLimit int) *FileProvider {
+	return &FileProvider{
+		sources:    sources,
+		filesLimit: filesLimit,
+	}
+}
+
+// FilesToTail returns all the Files matching paths in sources,
+// it cannot return more than filesLimit Files.
+// For now, there is no way to prioritize specific Files over others
+// and the result is not meant to remain the same over serveral calls
+func (r *FileProvider) FilesToTail() []*File {
+	filesToTail := []*File{}
+	for _, source := range r.sources {
+		// search all files matching pattern and append them all until filesLimit is reached
+		pattern := source.Path
+		paths, err := filepath.Glob(pattern)
+		if err != nil {
+			log.Println("Could not find any file matching pattern", pattern)
```

Comment:
[#Glob](https://golang.org/pkg/path/filepath/#Glob) documentation says:
> Glob ignores file system errors such as I/O errors reading directories. The only possible returned error is ErrBadPattern, when pattern is malformed.
So I think this message needs to be changed.
Otherwise, good change.